File: LinkedLists/smruthi/LL.py
class Node:
    def __init__(self,value):
        self.data=value
        self.next=None

def find_max(head):
        if not head:
            return None
        mv=head.data
        cur=head.next
        
        while cur:
            if cur.data > mv:
                mv=cur.data
            cur=cur.next
        return mv

def reverse_inplace(head):
    if not head :
        return None
    prev =None
    cur = head
    while cur:
        temp=cur.next
        cur.next=prev
        prev=cur
        cur=temp
    return prev
     

head=Node(10)
head.next=Node(50)
head.next.next=Node(15)
head.next.next.next=Node(25)
head.next.next.next.next=Node(1000)

temp=head
while temp :
    print(temp.data)
    temp=temp.next

print("Max of all the numbers in the LinkedList is ",find_max(head))
head = reverse_inplace(head)
print("The reverse of the Linked List :")
temp=head
while temp:
    print(temp.data)
    temp=temp.next
 

File: LinkedLists/yashas/ll.java
#HW1
class Node{
    int data;
    Node next;

    Node(int d,Node n){
        data=d;
        next=n;
    }
    Node(int d2){
        data=d2;
        next=null;
    }
}
public class ll{
    public static Node reverse(Node n){
    Node prev=null;
    Node temp=n;
    while(temp!=null){
        Node front=temp.next;
        temp.next=prev;
        prev=temp;
        temp=front;
    }
    return prev;
}
public static int max(Node n){
    Node temp=n;
    int maximum=Integer.MIN_VALUE;
    if(n==null) return Integer.MIN_VALUE;
    while(temp!=null){
        maximum=Math.max(maximum,temp.data);
        temp=temp.next;
    }
    return maximum;
}
    public static void main(String [] args){
        Node head = new Node(10);
        head.next = new Node(20);
        head.next.next = new Node(30);
        head.next.next.next = new Node(40);
        head.next.next.next.next = new Node(50);
        System.out.println("MAX= "+max(head));
        Node res=reverse(head);
        Node temp = res;
        while (temp!=null) {
            System.out.print(temp.data + " ");
            temp = temp.next;
        }
    }
}
